vm/runtime/src/lib.rs

```diff
@@ -211,9 +215,46 @@ pub trait Syscalls {
         Ok(())
     }
     /// Verifies a proof of spacetime.
-    fn verify_post(&self, _vi: &WindowPoStVerifyInfo) -> Result<(), Box<dyn StdError>> {
-        // TODO
-        todo!()
+    fn verify_post(&self, verify_info: &WindowPoStVerifyInfo) -> Result<(), Box<dyn StdError>> {
+        type ReplicaMapResult = Result<(SectorId, PublicReplicaInfo), String>;
+
+        //collect registered proofs
+        let registered_proofs = &verify_info
+            .proofs
+            .iter()
+            .map(|p| p.registered_proof as u8)
+            .collect::<Vec<_>>();
+
+        //collect replicas
+        let replicas = verify_info
+            .private_proof
```

Comment:
And also do you mind updating this field name, it was updated in specs-actors after we setup the type (easier to follow if we match names)
```suggestion
            .challenged_sectors
```

Reply:
Sweet will update it thanks!